azuredevops/internal/acceptancetests/testutils/commons.go

```diff
@@ -51,8 +51,12 @@ func PreCheck(t *testing.T, additionalEnvVars *[]string) {
 	}
 
 	for _, variable := range requiredEnvVars {
-		if strings.EqualFold(os.Getenv(variable), "") {
-			t.Fatalf("`%s` must be set for acceptance tests!", variable)
+		if _, ok := os.LookupEnv(variable); !ok {
+			// Skip tests with missing additional environment variables when SKIP_NOT_SET_TESTS set
+			if _, ok := os.LookupEnv("SKIP_NOT_SET_TESTS"); ok && additionalEnvVars != nil {
```

Comment:
@EliiseS  Can you provide your background concerns here. I cannot get the point why add `SKIP_NOT_SET_TESTS`.
For example: we set `AZDO_PERSONAL_ACCESS_TOKEN`, `AZDO_ORG_SERVICE_URL` and `SKIP_NOT_SET_TESTS`. If we run `make testacc`, all service connections and other  acceptance tests relies on environment configuration will be ignored  because they need the environment configures (constraint: configurations are not set in .tf file).

Reply:
As you say correctly, contributors only focus on the one resource they want to contribute. If a developer now wants to run acceptance tests locally, it would be a great simplification if the tests for which no infrastructure/token/account is available (e.g. Bitbucket) could simply be disabled. 

We already have a solution here using the build tags in the `_test.go` files and the PowerShell script `acctest.ps1` but maybe we need a more simple solution. I think the easier it is for the provider to develop, the more likely it is that there will be many people who want to do it. The entry hurdles must not be too high.

Reply:
we can run the specify acceptance test with 
`make testacc TESTARGS='-run=TestAccServiceEndpointBitBucket' `   or
`TF_ACC=1 go test -tags "all" ./... -run=TestAccServiceEndpointBitBucket -v  -timeout 120m`
`-run` param will match test case with function name prefix.

Reply:
@xuzhang3  I agree, especially because @EliiseS added with 77a7add4442fba55a55178f3a0e40dd6642da8c9 the possibility to specify build tags for the `GNUMakefile` (as you already can with the `acctests.ps1` script) to limit (exclude) the number of acceptance tests being run. So I think we can then do without `SKIP_NOT_SET_TESTS`.

Reply:
I added this because I'd like to run all tests locally, and skip `specialty` tests I haven't set up yet, such as ones requiring 	`AZDO_DOCKERREGISTRY_SERVICE_CONNECTION_USERNAME`, `AZDO_DOCKERREGISTRY_SERVICE_CONNECTION_EMAIL`, `AZDO_DOCKERREGISTRY_SERVICE_CONNECTION_PASSWORD`, ect.

Basically when I have to run acceptance tests manually locally.

Reply:
@EliiseS yes that's really convenient but as I said, after you added 77a7add, you can limit the acceptance test to execute by specifying their build tags (or exclusion tags). So basically after 77a7add there isn't a real need for `SKIP_NOT_SET_TESTS`. I would be fine having both possibilities here but @xuzhang3 is more strict on such things I guess.

Reply:
Actually, Xu and I have spoken and he's enlightened me to the fact that almost all of the values used in the acceptance tests can be set as dummy values

Reply:
But the dummy values will make the tests fail as if the environment variables aren't set. Right?

Reply:
No, all the tests pass. (I know, I was shocked too)

Reply:
WTF!?!?!? 😮  What nonsense! 🤷‍♂️ 🤦‍♂️